src/main/java/teammates/common/util/LogMessageGenerator.java

```diff
@@ -0,0 +1,252 @@
+package teammates.common.util;
+
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Map;
+import java.util.TimeZone;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import teammates.common.datatransfer.UserType;
+import teammates.common.datatransfer.attributes.AccountAttributes;
+import teammates.common.datatransfer.attributes.StudentAttributes;
+import teammates.common.exception.TeammatesException;
+import teammates.common.util.ActivityLogEntry.Builder;
+
+/**
+ * Factory to generate log message.
+ */
+public class LogMessageGenerator {
+    public static final Pattern PATTERN_ACTION_NAME = Pattern.compile("^/\\S+?/(?<actionName>[^\\s\\?]*)");
+    public static final String PATTERN_ACTION_NAME_GROUP = "actionName";
+
+    /**
+     * Generates log message for all *Action page.
+     *
+     * @param url URL of the action
+     * @param params parameterMap of the request
+     * @param currUser login information generated by {@link teammates.logic.api.GateKeeper}
+     * @param userAccount authentication user account generated by action
+     * @param unregisteredStudent authentication unregisteredStudent attributes generated by action
+     * @param logMessage log message to show to admin
+     * @return log message in the form specified in {@link ActivityLogEntry}
+     */
+    public String generatePageActionLogMessage(String url, Map<String, String[]> params, UserType currUser,
+            AccountAttributes userAccount, StudentAttributes unregisteredStudent, String logMessage) {
+        Builder builder = generateBasicActivityLogEntry(url, params, currUser);
```

Comment:
Sounds weird as we have four public APIs in `LogMessageGenerator` to generator different log messages.